crates/terminal_view/src/terminal_view.rs

```diff
@@ -859,14 +867,18 @@ impl TerminalView {
     }
 
     fn focus_in(&mut self, cx: &mut ViewContext<Self>) {
-        self.terminal.read(cx).focus_in();
+        self.terminal.update(cx, |terminal, _| {
+            terminal.set_cursor_style(self.cursor_style);
+            terminal.focus_in();
+        });
         self.blink_cursors(self.blink_epoch, cx);
         cx.notify();
     }
 
     fn focus_out(&mut self, cx: &mut ViewContext<Self>) {
         self.terminal.update(cx, |terminal, _| {
             terminal.focus_out();
+            terminal.set_cursor_style(CursorStyle::HollowBlock);
```

Comment:
Just for my own understanding: wasn't this the case before? Why do we have to add this?